prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,229 @@
+## Proposal for Code-to-Condition Mapping
+
+## Background
+
+Public Health reporting in the United States is not uniform in nature with each public health agency (PHA) setting their
+own criteria for what conditions need to be reported to that jurisdiction. In the context of electronic lab reporting (ELR), the onus is typically on the reporting entity (lab, hospital, clinic etc.) to determine which result messages qualify for each PHA the entity reports to.
+
+Since ReportStream takes on the burden of identifying the appropriate destination for sender’s messages, it will be necessary to implement a feature to ensure only messages for the appropriate conditions/diseases are allowed to route to each connected receiver. This feature will be similar to the COVID-19 LIVD table in that it will be a table of LOINC codes and their associated conditions that is used to reference received LOINC/SNOMED codes and return their associated condition.
+
+## Assumptions
+1.) The CSTE ValueSets that make up the Reportable Conditions Trigger Codes (RCTC) will be sufficient to capture a majority of codes sent to ReportStream. If this turns out not to be true we can modify the design to include additional data sources.
+
+## Criteria
+
+1.) Must be able to map LOINC and SNOMED codes to condition.<br>
+2.) Tables must have the ability to be updated without a PR.<br>
+3.) Mapping must account for all 160 conditions available in Report Content Knowledge Management System that relate to ELR (https://www.rckms.org/conditions-available-in-rckms/) <br>
+4.) Must be able to add ad-hoc mappings as needed for local codes/LDTs or other non-standard codes
+
+## Out of Scope
+The below items are not necessary to accomplish the main task of mapping code-to-condition but are instead enhancements to the core functionality that will have their own work effort.
+
+1.) Utility to check sender test compendiums against current mapping table<br>
+2.) Utility to automatically update tables from Value Set Authority Center API (https://www.nlm.nih.gov/vsac/support/usingvsac/vsacfhirapi.html) <br>
+3.) Reporting mechanism to inform RS/sender when unmapped value was found.<br>
+4.) Mechanism to automate which conditions each state considers reportable.
+
+## Design
+
+
+### Condition Mapping Table
```

Comment:
Would there need to be new functionality to update these tables without deployments or does the existing lookup code logic cover that?

Reply:
The existing CLI tools allow for tables to be updated at any time